### ISSUE TITLE
nl80211: Driver does not support authentication/association or connect commands

### ISSUE THREAD

**Issue by BenSapiens** [desflynn/realtek-8192cu-concurrent-softAP#2]
Hey, thanks for the guide, it's absolutely brilliant. I would've never figured any of this out myself. If I may appeal to your expertise, I have a quick question.

I have a wifi dongle with chipset "Realtek RTL8188CUS". With your excellent guide, I've successfully gotten concurrent station+AP working — mostly.

The title describes the (fatal) error I get when I attempt to connect to my wifi router via `wpa_supplicant -i wlan0 -c /etc/wpa_supplicant/wlan0.conf`. When I try it with `-D 8192cu`, the module name that appears in `lsmod`, I get the (fatal) error `wlan0: Unsupported driver '8192cu'` — same thing for `rtl871xdrv`, as found in `hostapd.conf`. When I try with `-D wext`, I get a successful connection.

Is "wext" somehow resolving to "8192cu"?

Does `wpa-driver wext` in `/etc/network/interfaces` work for you?

Is it normal to have the wlan1 AP operate in only b/g mode and give a (fatal) error when `hw_mode=g` in `hostapd.conf` is changed to `hw_mode=n`?

Many thanks.

**Comment by BenSapiens** [comment]
I've solved my problem by bypassing `/etc/network/interfaces`, but I still have no idea what's going on with the driver and I don't know why `/etc/network/interfaces` isn't working properly with `wpa_supplicant`. `ctrl_interface=DIR=/var/run/wpa_supplicant` seems to have no effect, even with a manually created directory with 777 permissions.

I'm still a bit curious about what's going on with `wext`.

Here's how I patched over the problem: I duplicated `wpa_supplicant.service` in `/lib/systemd/system` to a new file called `wlan0.service` with an `ExectStart=` line as follows:

`/sbin/wpa_supplicant -u -s -g /run/wpa_supplicant/sock -i wlan0 -D wext -c /etc/wpa_supplicant/wlan0.conf`

Then I started and enabled the service, and now it works on boot.

There are still other errors. From wpa_supplicant I'm seeing:

`wpa_supplicant[18083]: ioctl[SIOCSIWAP]: Operation not permitted`
`wpa_supplicant[18083]: ioctl[SIOCSIWENCODEEXT]: Invalid argument`
`wpa_supplicant[18083]: ioctl[SIOCSIWENCODEEXT]: Invalid argument`

...immediately upon starting the service, and when it connects to the AP, I see:

`wpa_supplicant[18083]: wlan0: Association request to the driver failed`

...but right after that:

`wpa_supplicant[18083]: wlan0: Associated with e0:3f:49:9d:a3:c0`

And it seems to work fine, so I'm not complaining.

I'm running Raspbian, by the way.

Sorry if this isn't the right place for the previous post and this one, but I think it should at least be out there somewhere in case somebody else has a problem like this.

**Comment by BenSapiens** [comment]
I figure I should comment before I close this.

In summary, I woke up today and it's stopped working entirely. There are no discernible errors in any of the `systemctl status service` outputs and nothing out of the ordinary in the `syslog`. I've now spent probably 4 hours on this thing in total, which is just too much for something which I have no idea how it works and little interest in finding out. $35 / 4 = 8.75. So, I ordered a Pi 3 from Amazon.

Thanks for your effort, truly, but I need to know that my tech won't stop working at random intervals or upon future software updates.
